src/data/equiv/ring.lean

```diff
@@ -282,6 +282,34 @@ noncomputable def of_bijective (f : R →+* S) (hf : function.bijective f) : R 
 lemma of_bijective_apply (f : R →+* S) (hf : function.bijective f) (x : R) :
   of_bijective f hf x = f x := rfl
 
+/-- A family of ring isomorphisms `Π j, (R j ≃+* S j)` generates a
+ring isomorphisms between `Π j, R j` and `Π j, S j`.
+
+This is the `ring_equiv` version of `equiv.Pi_congr_right`, and the dependent version of
+`ring_equiv.arrow_congr`.
+-/
+@[simps]
+def Pi_congr_right {ι : Type*} {R S : ι → Type*}
+  [Π i, semiring (R i)] [Π i, semiring (S i)]
+  (e : ∀ i, R i ≃+* S i) : (Π i, R i) ≃+* Π i, S i :=
+{ .. @mul_equiv.Pi_congr_right ι R S _ _ (λ i, (e i).to_mul_equiv),
```

Comment:
Does `simps` generate something nice? if not, you might want to add
```suggestion
{ to_fun := λ x j, es j (x j),
  inv_fun := λ x j, (es j).symm (x j),
  .. @mul_equiv.Pi_congr_right ι R S _ _ (λ i, (e i).to_mul_equiv),
```

Reply:
I think `simps` does the right thing here because `mul_equiv.Pi_congr_right` also has a `simps` lemma. But this can't hurt.